### ISSUE TITLE
hardswish_grad is built from relu6_grad, whose indicator is open, so the hard Swish gradient is wrong at exactly x = ±3

### ISSUE THREAD

**Issue by truong-v** [BobMcDear/attorch#16]
## Summary

ATen's `hardswish_backward` is `0` for `x < -3`, `x/3 + 0.5` on the **closed** `[-3, 3]`, and `1` for `x > 3`. attorch composes the gradient out of `relu6_grad`, whose indicator `(0 < input) & (input < 6)` is open at both ends, so the `input * relu6_grad(input + 3)` term is silently dropped at `x = -3` and `x = +3`.

| x | torch | attorch |
|---|---|---|
| -3 | -0.5 | 0.0 |
| +3 | 1.5 | 1.0 |

The forward is correct at both points, so this is backward-only, and `relu6_grad` itself is **not** wrong — ATen's `hardtanh_backward` is 0 at both ends, so the open indicator is exactly right for `ReLU6`; it is only the reuse in `hardswish_grad`, where the band needs to be closed, that is wrong. `x = ±3` is not a contrived value either: it is the knee of hard Swish, `nn.Hardswish` is the activation in MobileNetV3-class networks, and the boundary is reached in practice — `tests/test_act_layers.py` fails 17 `Hardswish` parametrizations today and `tests/test_glu_layer.py` one more, all at the larger shapes, where with millions of `randn` samples one eventually lands exactly on ±3.

## Reproduction

```python
import torch
from torch import nn

import attorch

x = torch.tensor([-4.0, -3.0, -1.0, 0.0, 1.0, 3.0, 4.0], device='cuda')
row = lambda t: '  '.join(f'{v:8.4f}' for v in t.tolist())

for name, module in (('torch  ', nn.Hardswish()), ('attorch', attorch.Hardswish())):
    xx = x.clone().requires_grad_(True)
    y = module(xx)
    y.backward(torch.ones_like(y))
    print(f'{name} forward ', row(y.detach()))
    print(f'{name} gradient', row(xx.grad))
```

```
torch   forward   -0.0000   -0.0000   -0.3333    0.0000    0.6667    3.0000    4.0000
torch   gradient   0.0000   -0.5000    0.1667    0.5000    0.8333    1.5000    1.0000
attorch forward   -0.0000   -0.0000   -0.3333    0.0000    0.6667    3.0000    4.0000
attorch gradient   0.0000    0.0000    0.1667    0.5000    0.8333    1.0000    1.0000
```

Every interior point agrees.

## Environment

- attorch at `88ed34d`
- NVIDIA L40S (sm_89)
- torch 2.4.0+cu121, triton 3.0.0, Python 3.10, driver CUDA 12.4
